backend/de.metas.handlingunits.base/src/test/java/org/adempiere/mm/attributes/api/impl/SubProducerAttributeDAOTest.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * #%L
+ * de.metas.handlingunits.base
+ * %%
+ * Copyright (C) 2021 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package org.adempiere.mm.attributes.api.impl;
+
+import de.metas.util.Services;
+import org.adempiere.mm.attributes.api.ISubProducerAttributeDAO;
+import org.adempiere.service.ClientId;
+import org.adempiere.test.AdempiereTestHelper;
+import de.metas.interfaces.I_C_BP_Relation;
+import org.compiere.model.I_C_BPartner;
+import org.compiere.model.X_C_BP_Relation;
+import org.compiere.util.Env;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.Properties;
+
+import static org.adempiere.model.InterfaceWrapperHelper.newInstance;
+import static org.adempiere.model.InterfaceWrapperHelper.save;
+import static org.assertj.core.api.Assertions.*;
+
+public class SubProducerAttributeDAOTest
+{
+	private I_C_BPartner bp;
+	private I_C_BPartner bp2;
+	private final ISubProducerAttributeDAO dao = Services.get(ISubProducerAttributeDAO.class);
```

Comment:
this might be the reason for the test failures. suggestion: set the dao in the init-method, after `AdempiereTestHelper.get().init();`

Reply:
PSA: this is not the problem, but still please create a fresh `dao` for each test. Just as a rule of thunb